public/components/trace_analytics/requests/traces_request_handler.ts

```diff
@@ -472,6 +324,31 @@ const hitsToSpanDetailData = async (hits: any, colorMap: any, mode: TraceAnalyti
   return data;
 };
 
+interface Hit {
+  _index: string;
+  _id: string;
+  _score: number;
+  _source: any;
+  sort?: any[];
+}
+
+interface ParsedResponse {
+  hits?: {
+    hits: Hit[];
+  };
+  [key: string]: any;
+}
+
+export function normalizePayload(parsed: ParsedResponse): Hit[] {
+  if (Array.isArray(parsed)) {
+    return parsed;
+  }
+  if (parsed.hits && Array.isArray(parsed.hits.hits)) {
+    return parsed.hits.hits;
+  }
+  return [];
+}
```

Comment:
Why do we still need this? If we know one check is for Jaeger and other is data prepper can we just use mode based if conditions instead?

Reply:
Updated too 
```
export function normalizePayload(parsed: ParsedResponse): Hit[] {
  if (parsed.hits && Array.isArray(parsed.hits.hits)) {
    return parsed.hits.hits;
  }
  return [];
}
```
Jaeger and data prepper should be under parsed.hits